File: frontend/src/types/reminder.ts
export interface Reminder {
  id: string;
  task_id?: string;
  vendor_id?: string;
  title: string;
  message?: string;
  reminder_type: ReminderType;
  remind_at: string;
  recurrence: Recurrence;
  notification_channels: string; // JSON string: ["browser", "email"]
  status: ReminderStatus;
  created_at: string;
  updated_at: string;
}

export type ReminderType = 'follow_up' | 'payment_due' | 'meeting' | 'deadline' | 'custom';
export type ReminderStatus = 'pending' | 'sent' | 'dismissed' | 'snoozed';
export type Recurrence = 'none' | 'daily' | 'weekly' | 'monthly';

export interface CreateReminderRequest {
  task_id?: string;
  vendor_id?: string;
  title: string;
  message?: string;
  reminder_type: ReminderType;
  remind_at: string;
  recurrence?: Recurrence;
  notification_channels: string[]; // Array that will be JSON stringified
  status?: ReminderStatus;
}

export interface UpdateReminderRequest {
  task_id?: string;
  vendor_id?: string;
  title?: string;
  message?: string;
  reminder_type?: ReminderType;
  remind_at?: string;
  recurrence?: Recurrence;
  notification_channels?: string[]; // Array that will be JSON stringified
  status?: ReminderStatus;
}

File: backend/internal/models/reminder.go
package models

import (
	"time"
)

// Reminder represents a reminder for a task or vendor
type Reminder struct {
	ID                  string     `json:"id"`
	TaskID              *string    `json:"task_id,omitempty"`
	VendorID            *string    `json:"vendor_id,omitempty"`
	Title               string     `json:"title" binding:"required"`
	Message             *string    `json:"message,omitempty"`
	ReminderType        string     `json:"reminder_type" binding:"required"`
	RemindAt           time.Time  `json:"remind_at" binding:"required"`
	Recurrence          string     `json:"recurrence"`
	NotificationChannels string    `json:"notification_channels" binding:"required"` // JSON array
	Status              string     `json:"status"`
	CreatedAt           time.Time  `json:"created_at"`
	UpdatedAt           time.Time  `json:"updated_at"`
}

// CreateReminderRequest represents the request body for creating a reminder
type CreateReminderRequest struct {
	TaskID              *string   `json:"task_id"`
	VendorID            *string   `json:"vendor_id"`
	Title               string    `json:"title" binding:"required"`
	Message             *string   `json:"message"`
	ReminderType        string    `json:"reminder_type" binding:"required"`
	RemindAt            time.Time `json:"remind_at" binding:"required"`
	Recurrence          string    `json:"recurrence"`
	NotificationChannels []string  `json:"notification_channels" binding:"required"`
	Status              string    `json:"status"`
}

// UpdateReminderRequest represents the request body for updating a reminder
type UpdateReminderRequest struct {
	TaskID              *string   `json:"task_id"`
	VendorID            *string   `json:"vendor_id"`
	Title               *string   `json:"title"`
	Message             *string   `json:"message"`
	ReminderType        *string   `json:"reminder_type"`
	RemindAt            *time.Time `json:"remind_at"`
	Recurrence          *string   `json:"recurrence"`
	NotificationChannels *[]string `json:"notification_channels"`
	Status              *string   `json:"status"`
}

// ValidReminderTypes defines valid reminder types
var ValidReminderTypes = []string{
	"follow_up", "payment_due", "meeting", "deadline", "custom",
}

// ValidReminderStatuses defines valid reminder statuses
var ValidReminderStatuses = []string{
	"pending", "sent", "dismissed", "snoozed",
}

// ValidRecurrences defines valid recurrence patterns
var ValidRecurrences = []string{
	"none", "daily", "weekly", "monthly",
}

